lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs

```diff
@@ -118,46 +118,108 @@ public Stream Create(Uri fileUri)
         public void Delete(Uri fileUri)
         {
             // Delete the file via the hdfs command line.
-            _commandRunner.Run("dfs -rm " + fileUri.AbsolutePath);
+            try
+            {
+                _commandRunner.Run("dfs -rm " + fileUri);
+            }
+            catch
+            {
+                _commandRunner.Run("dfs -rm " + fileUri.OriginalString);
+            }
         }
 
         public bool Exists(Uri fileUri)
         {
             // This determines the existence of a file based on the 'ls' command. 
             // Ideally, we'd use the 'test' command's return value, but we did not find a way to access that.
-            return
-                _commandRunner.Run("dfs -ls " + fileUri.AbsolutePath).StdErr
-                    .All(line => !NoSuchFileOrDirectoryRegEx.IsMatch(line));
+            try
+            {
+                return
+                     _commandRunner.Run("dfs -ls " + fileUri).StdErr
+                         .All(line => !NoSuchFileOrDirectoryRegEx.IsMatch(line));
+            }
+            catch
```

Comment:
Same as above.

Reply:
The change in REEF-1827 actually breaks our client after they pick us that REFF change. The file was uploaded but downstream is not able to find the file. The reason is because the file extension is slightly different from original.

Reply:
@jwang98052: The file extension changes? What was the old, what is the new?